pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -457,7 +457,7 @@ func buildSidecarListenerTLSContext(authenticationPolicy *authn.Policy) *auth.Do
 				AlpnProtocols: ListenersALPNProtocols,
 			},
 			RequireClientCertificate: &google_protobuf.BoolValue{
-				Value: !mTLSParams.AllowTls,
+				Value: !(mTLSParams != nil && mTLSParams.AllowTls),
```

Comment:
I saw this code but the reason I didn't fix was coz I thought requireTLS should be false if global mesh.Auth is set to none. Shouldn't that be the case? Otherwise, this code will still fail, because we are setting server side certificates in Envoy, but these don't exist and clients will still talk without TLS.

Reply:
Not really, the policy can come from per-service annotation or actual authn policy (CRD). I believe the test is enable mTLS per service, so it still get inside the if.

This may also related to this https://github.com/istio/istio/issues/4763, which cause mTLSParam is nil instead of being default message.

Reply:
Ah okay.